datafusion/expr-common/src/type_coercion/binary.rs

```diff
@@ -1138,27 +1138,44 @@ fn numeric_string_coercion(lhs_type: &DataType, rhs_type: &DataType) -> Option<D
     }
 }
 
+fn coerce_list_children(lhs_field: &FieldRef, rhs_field: &FieldRef) -> Option<FieldRef> {
+    Some(Arc::new(
+        Arc::unwrap_or_clone(Arc::clone(lhs_field)).with_data_type(comparison_coercion(
```

Comment:
@alamb can there be a problem if we coerce two lists and they have different field _names_?

@blaginin the resulting field nullability should be OR of nullability of sources

Reply:
https://github.com/apache/datafusion/pull/13468 seems related to this

Reply:
Given that this PR doesn’t change names behaviour, let’s go as is and then fix it separately in that PR you highlighted?

Reply:
i am fine assuming the answer to this question is 'nope':

_can there be a problem if we coerce two lists and they have different field names?_

i hope it is 'nope, no problem'
@alamb @jayzhan211 please confirm

Reply:
I don't think there is any problem in theory. I am a little fuzzy about what the semantic meaning of the Field's name in a `DataType::List` means -- sometimes the code is overly pedantic when comparing but I think semantically two lists are the same if their element's types are the same (and nullness and metadata). I don't think the field name should be compared

However, I remember there have been issues before on this point

Reply:
I think https://github.com/apache/datafusion/issues/13481 is a complain that list item name isn't preserved in some cases. I don't know when it would matter though.  From SQL perspective, it shouldn't.